src/server/auth.ts

```diff
@@ -43,6 +44,46 @@ passport.use(
 	)
 );
 
+// create hacker from user
+export const createHackerFromUser = async (user: InstanceType<User>, done: Function) => {
+	const createdHacker = await HackerModel.create({
+		email: user.email,
+		status: Status.Created,
+		user: user._id,
+	});
+	if (createdHacker) {
+		logger.debug(user);
+		done(null, user);
+	} else {
+		done(null, false);
+	}
+};
+
+// create a new user
+export const createAuthenticatedUser = async (
+	authType: AuthType,
+	email: string,
+	id: any,
+	done: Function
+) => {
+	logger.debug('> Creating user.....');
+	const newUser = {
+		authLevel: AuthLevel.HACKER,
+		authType,
+		email,
+		githubId: authType === AuthType.GITHUB ? id : undefined,
```

Comment:
I'm totally cool if we want to do this work later but it might be wise for us to get flexible with more OAuth providers now.

Reply:
yeah another issue